homeassistant/components/switch/insteon_plm.py

```diff
@@ -19,34 +19,43 @@
 @asyncio.coroutine
 def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up the INSTEON PLM device class for the hass platform."""
+
+    state_list = []
     plm = hass.data['insteon_plm']
 
-    device_list = []
-    for device in discovery_info:
-        name = device.get('address')
-        address = device.get('address_hex')
+    for deviceInfo in discovery_info:
+        address = deviceInfo['address']
+        device = plm.devices[address]
+        stateKey = deviceInfo['stateKey']
+        newnames = deviceInfo['newnames']
 
-        _LOGGER.info('Registered %s with switch platform.', name)
+        stateName = device.states[stateKey].name
 
-        device_list.append(
-            InsteonPLMSwitchDevice(hass, plm, address, name)
-        )
+        if stateName in ['lightOnOff', 'outletTopOnOff', 'outletBottomOnOff']:
```

Comment:
Why is this called name but used as a type?

Reply:
stateName is from the underlying `insteonplm` module. I agree the term may be confusing in this context but it is the only way to map the functionality of the HA device correctly.